### PR TITLE
[PORT+IMP][10.0] product_stock_state

### PR DESCRIPTION
**Pull request by legalsylvain**
See readme.

* remove sale_ok from the stock state computation. (will create a new sale_state, in a new module).
* Improve algorithm
* remove hardcoded value for the threshold of 10. threshold can be now set by company (sale settings) by category, or by product.

* OCA convention. (file name, description, screenshot, tests, etc...)

### CODE REVIEW

**Comment by legalsylvain** [comment]
please review first #63 . (trivial fix) to #MakeTravisGreenAgain.

File: product_stock_state/README.rst [hunk 1]
```diff
@@ -2,33 +2,49 @@
...
 
-========================
+===================
 Product Stock State
-========================
+===================
 
-This module add a stock state on the product in order to give an human readable information
-without giving the current quantity in stock
+This module add a stock state on the product in order to give an human readable
```
Comment (bealdav) [inline product_stock_state/README.rst:9]: s/an/a

File: product_stock_state/README.rst [hunk 2]
```diff
@@ -2,33 +2,49 @@
...
+  the new group 'Stock State Threshold by Product'.)
 
 Usage
 =====
 
-Go on the product tree/form and see the stock state
+Go on the product tree and see the stock state
+
+.. image:: /product_stock_state/static/description/product_product_tree.png
+     :width: 800 px
```
Comment (bealdav) [inline product_stock_state/README.rst:40]: TIL the real good way to define image in rst : image appears in githhub and in Odoo. Thanks you
/ + module_name
👍

File: product_stock_state/models/__init__.py [hunk 3]
```diff
@@ -1,3 +1,8 @@
 # -*- coding: utf-8 -*-
 
-from . import product
+from . import res_company
+from . import sale_config_settings
```
Comment (bealdav) [inline product_stock_state/models/__init__.py:4]: FYI, in v11 all settings are in the same model.
Not blocking, but for later PR we could call this file config_settings.py, then migration'll be smarter

**Review by bealdav — APPROVED** [review]

File: base_url/models/url_url.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+#    @author EBII MonsieurB <monsieurb@saaslys.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+import logging
+
+_logger = logging.getLogger(__name__)
+
+
+def get_model_ref(record):
```
Comment (lmignon) [inline base_url/models/url_url.py:12]: @legalsylvain The same method is already defined into abstract_url.py

**Comment by legalsylvain** [comment]
Hi. I realized this PR during the OpenDays 2017, and I'm sorry but I have no time for the time being to work on that PR.

but feel free to finish it, forking my branch !

kind regards.

File: product_stock_state/models/product_product.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        ('in_stock', 'In Stock'),
+        ('in_limited_stock', 'In Limited Stock'),
+        ('resupplying', 'Resupplying'),
+        ('out_of_stock', 'Out Of Stock'),
+    ]
+
+    stock_state = fields.Selection(
+        selection=_STOCK_STATE_SELECTION, compute='_compute_stock_state')
+
+    def _compute_stock_state(self):
```
Comment (lmignon) [inline product_stock_state/models/product_product.py:24]: @legalsylvain No `api.depends('qty_available', 'virtual_available')` ?

**Review by lmignon — APPROVED** [review]
Some minor non blocking comments otherwise LGTM

**Pull request merged**